src/api/install.cpp

```diff
@@ -334,11 +330,11 @@ namespace mamba
 
         std::vector<std::shared_ptr<MSubdirData>> subdirs;
         MultiDownloadTarget multi_dl;
-        std::unique_ptr<LockFile> subdir_download_lock;
+
+        std::vector<std::unique_ptr<LockFile>> subdir_download_locks;
         if (!ctx.offline)
-        {
-            subdir_download_lock = std::make_unique<LockFile>(cache_dir / "mamba.lock");
-        }
+            for (const auto& c : ctx.pkgs_dirs)
+                subdir_download_locks.push_back(std::make_unique<LockFile>(c / "mamba.lock"));
```

Comment:
do we really need to lock multiple? isn't it enough to lock the `first writable`?

Reply:
I removed that part to a clever lock, only when working with that cache (because we even don't know if the first writable one will be needed or not, or if the packages will be picked from one or multiple caches)